ts/Core/Axis/OrdinalAxis.ts

```diff
@@ -860,76 +965,37 @@ namespace OrdinalAxis {
          *
          * @return {number}
          */
-        axisProto.lin2val = function (
-            val: number,
-            fromIndex?: boolean
-        ): number {
+        axisProto.lin2val = function (val: number): number {
             let axis = this,
                 ordinal = axis.ordinal,
-                ordinalPositions = ordinal.positions,
-                ret;
+                ordinalPositions = ordinal.positions, // for the current visible range
+                ret = 0,
+                extendedOrdinalPositions = this.extendedOrdinalPositions;
 
-            // the visible range contains only equally spaced values
+            // The visible range contains only equally spaced values.
             if (!ordinalPositions) {
-                ret = val;
+                return val;
+            }
 
-            } else {
+            // If the value is not inside the plot area,
+            // use the extended positions.
+            // (array contains also points that are currently out of range).
 
-                let ordinalSlope = ordinal.slope,
-                    ordinalOffset = ordinal.offset,
-                    i = ordinalPositions.length - 1,
-                    linearEquivalentLeft,
-                    linearEquivalentRight,
-                    distance;
-
-
-                // Handle the case where we translate from the index directly,
-                // used only when panning an ordinal axis
-                if (fromIndex) {
-
-                    if (val < 0) { // out of range, in effect panning to the left
-                        val = ordinalPositions[0];
-                    } else if (val > i) { // out of range, panning to the right
-                        val = ordinalPositions[i];
-                    } else { // split it up
-                        i = Math.floor(val);
-                        distance = val - i; // the decimal
-                    }
+            // When iterating for the first time,
+            // get the extended ordinal positional and assign them.
+            if (axis.ordinal && !extendedOrdinalPositions) {
+                extendedOrdinalPositions = axis.ordinal.getExtendedPositions();
```

Comment:
Could we calculate all ordinal positions, only if it's not possible to translate `val` using `ordinalPositions`?